subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java

```diff
@@ -998,7 +1006,7 @@ public FileCollection getClasspath() {
     }
 
     public void setClasspath(FileCollection classpath) {
-        this.classpath = classpath;
+        this.classpath.setFrom(classpath);
```

Comment:
This may end up causing problems, when people do `test.classpath += something` and other similar . However, we want to move towards having a fixed identity for the classpath so perhaps let's make this change and see what the fallout is. We can either improve support for this kind of syntax or roll this change back and implement it in a different way (eg similar to what was done for `SourceTask`).

Reply:
I validated that `test.classpath += something` continues to work